oasislmf/model_preparation/summaries.py

```diff
@@ -562,7 +541,7 @@ def write_exposure_summary(
                     status,
                     loc_num
                 )
-            elif status != 'all':
+            elif status != 'all' and len(gul_inputs_errors_df) != 0:
```

Comment:
Just a small request - can you please replace `peril_code` with `peril_id` everywhere in this method? We are consistently using `peril_id` internally as the variable to store OED peril codes/IDs, and I think that is a better name.

The use of `peril` and `peril_code` is a bit confusing - you could maybe use `matching_peril_id` instead of `peril` in this statement

    peril = [v['desc'] for v in PERILS.values() if v['id'] == peril_code][0]

And you would use `peril_id` instead of `peril_code`, as mentioned above.

Reply:
Changed `peril` to `peril_desc` as it seemed a better fit.